From 7237fed84ec4e519d47b7cc59623c2bbfe748db2 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Thu, 30 Jul 2026 12:41:04 +0300
Subject: [PATCH 4/4] marvell: gbsyncd phy fix gbsyncd.service gearbox
 ConditionPathExists

Add Marvell Prestera external-PHY gbsyncd support.

PHYs are driven by a separate gbsyncd container (syncd -g 1, GB_ASIC_DB) using the
same mrvllibsai/libsai.so as main syncd, with gearbox_config.json, context_config.json,
and per-PHY hwsku JSON. Build with INCLUDE_GBSYNCD=y.

Omit ConditionPathExists=!gearbox_config.json on marvell-prestera when gearbox_config.json is present.

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 files/build_templates/per_namespace/gbsyncd.service.j2 | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/files/build_templates/per_namespace/gbsyncd.service.j2 b/files/build_templates/per_namespace/gbsyncd.service.j2
index ac695a6c..dea42bab 100644
--- a/files/build_templates/per_namespace/gbsyncd.service.j2
+++ b/files/build_templates/per_namespace/gbsyncd.service.j2
@@ -2,7 +2,11 @@
 [Unit]
 Description={{docker_container_name}} service
 Requires=database{% if use_instance_deps %}@%i{% endif %}.service config-setup.service
+{# ConditionPathExists=!...: gbsyncd does not start when gearbox_config.json exists (gearbox handled by gearsyncd in syncd). #}
+{# Platforms with a separate gbsyncd for external PHY omit this line when gearbox_config.json is present (see if below). #}
+{% if sonic_asic_platform != 'marvell-prestera' %}
 ConditionPathExists=!/usr/share/sonic/hwsku/gearbox_config.json
+{% endif %}
 After=database{% if use_instance_deps %}@%i{% endif %}.service config-setup.service
 After=interfaces-config.service
 After=swss{% if use_instance_deps %}@%i{% endif %}.service
-- 
2.25.1

